migration/migrator/migrations/course/20200111145749_office_hours_queue_2.py

```diff
@@ -0,0 +1,164 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+
+    # Remove old queue by just removing any old data
+    database.execute(
+        """
+        CREATE TABLE IF NOT EXISTS queue_new(
+        	entry_id SERIAL PRIMARY KEY,
+        	status TEXT NOT NULL,
+        	queue_code TEXT NOT NULL,
+        	user_id TEXT NOT NULL REFERENCES users(user_id),
+        	name TEXT NOT NULL,
+        	time_in TIMESTAMP NOT NULL,
+        	time_help_start TIMESTAMP,
+        	time_out TIMESTAMP,
+        	added_by TEXT REFERENCES users(user_id),
+        	help_started_by TEXT REFERENCES users(user_id),
+        	removed_by TEXT REFERENCES users(user_id)
+        );
+        """
+    )
+
+    # Old status code meanings
+    # const STATUS_CODE_IN_QUEUE = 0;//student is waiting in the queue
+    # const STATUS_CODE_BEING_HELPED = 1;//student is currently being helped
+    # const STATUS_CODE_SUCCESSFULLY_HELPED = 2;//student was successfully helped and is no longer in the queue
+    # const STATUS_CODE_REMOVED_BY_INSTRUCTOR = 3;//student was removed by the instructor
+    # const STATUS_CODE_REMOVED_THEMSELVES = 4;//student removed themselves from the queue
+    # const STATUS_CODE_BULK_REMOVED = 5;//student was removed after the empty queue button was pressed
+
+
+    # New status code meanings
+    # Status Code Key: "ABC" where ABC is a string of 3 numbers
+    #   A: who added you to the queue
+    #       0:Self
+    #       1:Mentor/TA/instructor
+    #   B: Current state in queue
+    #       0:Waiting
+    #       1:Being helped
+    #       2:Done/Fully out of the queue
+    #   C: Who removed you
+    #       0:Still in queue
+    #       1:Removed yourself
+    #       2:Mentor/TA helped you
+    #       3:Mentor/TA removed you
+    #       4:Kicked out because queue emptied
+    #       5:You helped you
```

Comment:
Let's break this apart into three columns instead of having it be contained in one column. It'll make it easier to potentially run future statistics over it, as well as just be a saner way of interacting with it.